File: Merge.py
def merge(arr, left, mid, right):
    while left <= mid and mid + 1 <= right:
        if arr[left] <= arr[mid + 1]:
            left += 1
        else:
            temp = arr[mid + 1]
            for i in range(mid, left - 1, -1):
                arr[i + 1] = arr[i]
            arr[left] = temp
            left += 1
            mid += 1

def merge_sort(arr, left, right):
    if left < right:
        mid = left + (right - left) // 2
        merge_sort(arr, left, mid)
        merge_sort(arr, mid + 1, right)
        merge(arr, left, mid, right)


arr = [5, 2, 8, 12, 25, 66, 32, 95, -2, 0]
print("Original array:", arr)

merge_sort(arr, 0, len(arr) - 1)
print("Sorted array:", arr)

File: Merge.java
public class MergeSort {
    public static void merge(int[] arr, int left, int mid, int right) {
        while (left <= mid && mid + 1 <= right) {
            if (arr[left] <= arr[mid + 1]) {
                left++;
            } else {
                int temp = arr[mid + 1];
                for (int i = mid; i >= left; i--) {
                    arr[i + 1] = arr[i];
                }
                arr[left] = temp;
                left++;
                mid++;
            }
        }
    }

    public static void mergeSort(int[] arr, int left, int right) {
        if (left < right) {
            int mid = left + (right - left) / 2;
            mergeSort(arr, left, mid);
            mergeSort(arr, mid + 1, right);
            merge(arr, left, mid, right);
        }
    }

    public static void main(String[] args) {
        int[] arr = {5, 2, 8, 12, 25, 66, 32, 95, -2, 0};
        System.out.print("Original array: ");
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();

        mergeSort(arr, 0, arr.length - 1);

        System.out.print("Sorted array: ");
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}
